### DISCUSSION TITLE
How to create custom relation type

### DISCUSSION THREAD

**Discussion by holoyan** [adonisjs/core#4399] [Help]
Is there a way to create custom relationship type rather then one-one, one-many many-many and so on.

For example in Laravel we can extend `Illuminate\Database\Eloquent\Relations\Relation` class and have our custom implementation, anything similar here?

**Comment by RomainLanz** [comment]
Hey @holoyan! 👋🏻 

We don't have documentation on it yet, but you can watch [our Q&A session recording](https://youtu.be/DJSrcb42koo?si=udJrA4sxyfH4UMt3&t=1533), where we discussed that.

**Comment by holoyan** [comment]
So I was digging into source code to understood what's going on, and seems there are some limitations,
If  I understood correct then to create custom relation first we need to define decorator [here](https://github.com/adonisjs/lucid/blob/develop/src/orm/decorators/index.ts) and [$addRelation](https://github.com/adonisjs/lucid/blob/develop/src/orm/base_model/index.ts#L432C11-L432C22) method accepts only this values - `hasOne,hasMany,belongsTo,manyToMany,hasManyThrough`

Please give me some direction if it's wrong

@RomainLanz @thetutlage
